script.js

```diff
@@ -0,0 +1,164 @@
+        return true;
+    }
+
+    function validateDetails(event){
+        event.preventDefault();
+        const existingusername=document.getElementById("existingusername").value;
+        const existingPassword=document.getElementById("existingPassword").value;
+        const userExist=userDetails.filter(obj=>obj.userEmail===existingusername);
+        const decriptPassword=crypt.decrypt(userExist[0].userPassword);
+        if (decriptPassword === existingPassword){
+            const user=userExist[0].userName;
+            const idString=userExist[0].id;
+            const userNameid="userName"+idString;
+            setCookie(userNameid,user,1);
+            document.getElementById("login").reset();
+            window.location.href = "https://aiswarya-a-m.github.io/login/dashboard.html";   
+        }
+        else{
+            alert("password is incorrect");
+        }
+    }
+    function usernameValid(username){
+        const usernamePattern=/^[a-zA-Z ]{3,30}$/;
+        if(usernamePattern.test(username)){
+            return true;
+        }else{
+            return false;
+        }
+    }
+    function emailValid(email){
+        const emailPattern=/^[a-zA-Z0-9._-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,4}$/;
+        if(emailPattern.test(email)){
+            return true;
+        }else{
+            return false;
+        }
+    }
+    function passwordValid(password){
+        if (/[!@#$%^&*()_+{}\[\]:;<>,.?~\-]/.test(password) &&
+        /[A-Z]/.test(password) &&
+        /\d/.test(password)
+        ) {
+            return true;
+        }
+        else{
+            return false;
+        }
+
+    }
+    function phoneNumberValid(phoneNumber){
+        const numberPattern= /^\d{10}$/;
+        if (numberPattern.test(phoneNumber)){
+            return true;
+        }else{
+            return false;
+        }
+    }
+    function emailNotExist(email){
+        const userExist=userDetails.findIndex(obj=>obj.userEmail===email)
+        if (userExist!==-1){
+            alert("email already exist")
+            return false;
+        }
+        else {
+            return true;
+        }
+    }
+    function phonenumNotExist(phoneNumber){
+        const userExist=userDetails.findIndex(obj=>obj.userNumber===phoneNumber)
+        if (userExist!==-1){
+            alert("phonenumber already exist")
+            return false;
+        }
+        else {
+            return true;
+        }
+    }
+
+    function setCookie(name,userNameid,dayslive){
+        const todate=new Date();
+        todate.setTime(todate.getTime()+(dayslive*24*60*60*1000));
+        let expiryDate="expires="+todate.toUTCString();
+        document.cookie=`${name}=${userNameid};${expiryDate}; path/`
+    }
+
+    function addDeatails(username,email,phoneNumber,password){
+        let id;
+        if (userDetails.length===0){
+            id=1
+        }else{
+            id=userDetails[userDetails.length-1].id+1
+        }   
```

Comment:
```suggestion
        let id = userDetails.length + 1;
```

`userDetails.length` will be zero if it's empty. so remove the `if/else` block for the id calculation and do this instead.